File: Data Structures & Algorithms/longest-common-prefix/submission-1.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        string s = strs[0];
        for(int i =1;i<strs.Length;i++){
            int j=0;
            while(j<s.Length && j<strs[i].Length && s[j] == strs[i][j])
                j++;
            s = s.Substring(0,j);
            if(s.Length == 0) return "";
        }
        return s;
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-1.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        result = []
        for s in strs:
            result.append(str(len(s))+'#'+s);
        return "".join(result);
    def decode(self, s: str) -> List[str]:
        n = len(s)
        i = 0
        arr = []
        while i<n:
            j = i
            while s[j] != '#':
                j += 1
            length = int(s[i:j])
            j+=1
            arr.append(s[j:j+length])
            i = j+length
        return arr
